kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -330,6 +331,41 @@ public static String concatWithDot(List<String> columnPath) {
     return columnPath.stream().map(SchemaUtils::escapeDots).collect(Collectors.joining("."));
   }
 
+  /**
+   * Traverse a schema
+   *
+   * @param schema root of the schema to be traversed
+   * @param callback callback on each StructField encountered
+   */
+  public static void traverse(StructType schema, BiConsumer<List<String>, StructField> callback) {
```

Comment:
We already have something called `SchemaIterable` to similar purposes. Please use it.